drivers: misc: Add APML driver support

abinayaddhandapani opened the pull request:
- Added Makefile for APML modules
- The driver implemenst APML alert and SBRMI, SBTSI functionalities
- Includes common SBRMI functions and headers
- Updated Kconfig and Makefile to include APML modules

Commit:
drivers: misc: Add SBRMI driver support

Updated Kconfig and Makefile to include SBRMI driver

This commit introduces the apml_sbrmi interface driver for BMC to support
monitoring and configuring power-related system parameters on AMD EPYC
processors via the APML interface.

The apml_sbrmi driver is ported from the apml_modules github repo
(https://github.com/amd/apml_modules/)

Key Features:
- Mailbox
- CPUID (Read-Only)
- MCA MSR (Read-Only)
- RMI Register (Read/Write)

Signed-off-by: Akshay Gupta <akshay.gupta@amd.com>
Signed-off-by: Abinaya Dhandapani <abinaya.dhandapani@amd.com>

ojayanth commented:
Please fix typo's and format issue in the commit message.
integrate apml_modules  mean you are integrating modules?  Here you are adding drivers right?

Add detailed commit message starting from this is ported from apml_modules repository and details of the driver.

please refer upstream  patch series in this area.  https://patchwork.kernel.org/project/linux-hwmon/cover/20240912070810.1644621-1-akshay.gupta@amd.com/

ojayanth commented on drivers/misc/Kconfig:
```diff
@@ -562,6 +562,14 @@ config TPS6594_PFSM
 	  This driver can also be built as a module.  If so, the module
 	  will be called tps6594-pfsm.
 
+config AMD_APML
```
any specific reason for Keeping single configuration option for  sbtsi sbrmi and alert feature?

I will be continuing the review once you finalized the direction.

ojayanth replied:
Better to split this in to multiple commit like sbtsi, sbrmi and contol by feature specific configuraion variables, instead on single.
Not enabling alert is really used by amd CRB 's now.

abinayaddhandapani replied:
Ack. The PR is split into two commits - SBTSI and SBRMI addition

ojayanth commented on drivers/misc/apml_modules/patches/0001-i3c-Export-struct-bus_type-i3c_bus_type-to-include-h.patch:
```diff
@@ -0,0 +1,57 @@
+From 1c1cc4879a67e28ad9fb080d4fa4383a525af7cd Mon Sep 17 00:00:00 2001
```
where is this patch used?

abinayaddhandapani replied:
This patch is not required when porting to linux-aspeed. Removing it.

ojayanth commented on drivers/misc/apml_modules/README.md:
```diff
@@ -0,0 +1,193 @@
+.. SPDX-License-Identifier: GPL-2.0
```
please add only relvent data , looks like this info is good for apml module based approch.

abinayaddhandapani replied:
Ack. Removed apml_alertl infos as it is not added as part of this porting

Commit:
drivers: misc: Add SBTSI driver support

Updated Kconfig and Makefile to include SBTSI driver

This commit introduces the apml_sbtsi interface driver for BMC to support
monitoring temperature related parameters on AMD EPYC processors
via the APML interface.

The apml_sbtsi driver is ported from the apml_modules github repo
(https://github.com/amd/apml_modules/)

Signed-off-by: Akshay Gupta <akshay.gupta@amd.com>
Signed-off-by: Abinaya Dhandapani <abinaya.dhandapani@amd.com>

abinayaddhandapani commented:
Ack. Added detailed commit message on both the commits.

ojayanth commented:
Please update PR details with updated scope.

abinayaddhandapani commented:
As apml-modules folder is renamed to amd-apml, facing issues with rebasing both the commits. 
Closing this pull request and will create a new PR.
